src/test/server/server_test_runner.cpp

```diff
@@ -45,6 +46,30 @@ class ServerTestRunner : public BaseTest {
     std::remove(_export_filename.c_str());
   }
 
+  bool file_exists(const std::string& name) {
+    std::ifstream file{name};
+    return file.good();
+  }
+
+  bool compare_files(const std::string& original_file, const std::string& created_file) {
+    std::ifstream original(original_file);
+    Assert(original.is_open(), "compare_file: Could not find file " + original_file);
+
+    std::ifstream created(created_file);
+    Assert(created.is_open(), "compare_file: Could not find file " + created_file);
+
+    std::istreambuf_iterator<char> iterator_original(original);
+    std::istreambuf_iterator<char> iterator_created(created);
+    std::istreambuf_iterator<char> end;
+
+    while (iterator_original != end && iterator_created != end) {
+      if (*iterator_original != *iterator_created) return false;
+      ++iterator_original;
+      ++iterator_created;
+    }
+    return ((iterator_original == end) && (iterator_created == end));
+  }
```

Comment:
Deduplicate and move to base_test? Maybe we should also move this and the existing implementations from base_test.hpp to base_test.cpp.

Reply:
I just moved all method implementations from base_test.hpp to base_test.cpp. The code compiles, but when running the test binary, I get errors like

```
dyld: lazy symbol binding failed: Symbol not found: __ZN7opossum17BaseTestWithParamIvE38get_supporting_segment_encodings_specsENS_6detail8DataTypeEb
  Referenced from: /Users/epic/h2/cmake-build-debug/./hyriseTest
  Expected in: flat namespace

dyld: Symbol not found: __ZN7opossum17BaseTestWithParamIvE38get_supporting_segment_encodings_specsENS_6detail8DataTypeEb
  Referenced from: /Users/epic/h2/cmake-build-debug/./hyriseTest
  Expected in: flat namespace
```

This occurs for different methods if I run different tests.

Reply:
When implementing templates in the cpp file you need to instantiate them. Can you get around using templates? For example, I don't see why `BaseTestWithParam<ParamType>::get_supporting_segment_encodings_specs` can't be free-floating.

Reply:
Okay, I thought there was a good reason for the methods being members of BaseTestWithParam. As none of them uses the ParamType, it would mean to exclude all of them (besides the destructor) from the class, right? If this is the case, should they rather be moved to a separate pair of cpp and hpp file that is included by base_test.hpp as they would not really have something in common anymore?

Reply:
Meh, I don't care too much about clean headers in the test. If it's a separate file, base_test should include it for convenience, so... whatever.

Reply:
Alright, I will not move the code to a separate header then.